provider/src/main/scala/com/actian/spark_vector/provider/RequestHandler.scala

```diff
@@ -123,8 +128,8 @@ class RequestHandler(sqlContext: SQLContext, val auth: ProviderAuth) extends Log
    * 2) return the extension of part.external_reference, if recognized
    * 3) throw IllegalArgumentException otherwise
    */
-  private def getFormat(part: JobPart): String =
-    part.format.orElse(PartialFunction.condOpt[String, String](part.external_reference.split("\\.").last) {
+  private def getFormat(part: JobPart): String = {
+    val ret = part.format.orElse(PartialFunction.condOpt[String, String](part.external_reference.split("\\.").last) {
       _ match {
```

Comment:
I might have missed this...but can we condense this match a little more?

`partFormat match {
   case "csv" | "parquet" | "json" | "orc" | "avro" =>
     checkFormat(partFormat)
     partFormat
 }`